frontend/packages/console-shared/src/utils/application-utils.ts

```diff
@@ -0,0 +1,77 @@
+import { k8sList, k8sPatch, K8sResourceKind, modelFor } from '@console/internal/module/k8s';
+import * as _ from 'lodash';
+
+const updateItemAppLabel = (item: K8sResourceKind, application) => {
+  const labels = _.cloneDeep(item.metadata.labels || {});
```

Comment:
Could this be:
```
  const labels = { ...item.metadata.labels, 'app.kubernetes.io/part-of': application || undefined };
```

If you still need to clone, it should be done after line 10.